precompile/assetsbridge/bridge_out.go

```diff
@@ -0,0 +1,443 @@
+
+	// now update the authorization to spend for the AssetsBridge
+	msg := banktypes.NewMsgSend(senderAddr.Bytes(), bridgeAddr.Bytes(), sdk.NewCoins(coin))
+	resp, err := sendAuth.Accept(context.SdkCtx(), msg)
+	if err != nil {
+		return fmt.Errorf("couldn't update authorization: %w", err)
+	}
+
+	if resp.Delete {
+		// Authorization fully consumed, delete it
+		err = m.authzKeeper.DeleteGrant(context.SdkCtx(), bridgeAddr, senderAddr, SendMsgURL)
+	} else if resp.Updated != nil {
+		err = m.authzKeeper.SaveGrant(context.SdkCtx(), bridgeAddr, senderAddr, resp.Updated, expiration)
+	}
+
+	if err != nil {
+		return fmt.Errorf("bridge is not authorized to burn BTC: %w", err)
+	}
+
+	if !resp.Accept {
+		return errors.New("bridge is not authorized to burn BTC")
+	}
+
+	if err := m.bridgeKeeper.BurnBTC(
+		context.SdkCtx(),
+		senderAddr.Bytes(),
+		sdkAmount,
+	); err != nil {
+		return fmt.Errorf("couldn't burn BTC: %w", err)
+	}
+
+	// finally update the journal entries to propagate the changes
+	// done to the gas token (BTC in our case)
+	balanceDelta, overflow := uint256.FromBig(inputs.Amount)
+	if overflow {
+		return fmt.Errorf("conversion from big.Int to uint256.Int overflowed: %v", inputs.Amount)
+	}
+
+	// only one side of the transfer to update here as we
+	// burnt funds
+	journal := context.Journal()
+	journal.SubBalance(common.BytesToAddress(senderAddr.Bytes()), balanceDelta, tracing.BalanceChangeTransfer)
+
+	return nil
+}
+
+// validate applies more extensive validation to the inputs, not only
+// the semantic validation but also verify balances, and tokens etc.
+func (m *BridgeOutMethod) validate(
+	context *precompile.RunContext,
+	inputs *bridgeOutInputs,
+) error {
+	sdkCtx := context.SdkCtx()
+
+	if inputs.Amount == nil {
+		return errors.New("amount is required")
+	}
+
+	if inputs.Amount.Sign() <= 0 {
+		return errors.New("amount must be positive")
+	}
+
+	_, ok := inputs.Chain.Validate()
+	if !ok {
+		return fmt.Errorf("unsupported chain: %v", inputs.Chain)
+	}
+
+	if err := m.validateRecipientForChain(inputs.Chain, inputs.Recipient); err != nil {
+		return err
+	}
+
+	// first check that the token is either BTC or a supported ERC20
+	if err := m.validateToken(sdkCtx, inputs.Token, inputs.Chain); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (m *BridgeOutMethod) validateToken(
+	sdkCtx sdk.Context,
+	token common.Address,
+	chain TargetChain,
+) error {
+	btcToken := common.HexToAddress(evmtypes.BTCTokenPrecompileAddress)
+	if bytes.Equal(btcToken.Bytes(), token.Bytes()) {
+		return nil
+	}
+
+	switch chain {
+	case TargetChainEthereum:
+		if _, ok := m.bridgeKeeper.GetERC20TokenMapping(sdkCtx, token.Bytes()); ok {
+			return nil
+		}
+
+		return fmt.Errorf("unsupported token: %v for ethereum target chain", token)
+	case TargetChainBitcoin:
+		return fmt.Errorf("unsupported token: %v for bitcoin target chain", token)
+	default:
+		panic("unreachable")
```

Comment:
Nitpick, but maybe a more precise message:
```suggestion
		panic("unreachable: unknown chain type")
```